pallets/parachain-staking/src/lib.rs

```diff
@@ -1160,4 +1158,26 @@ pub mod pallet {
 		// TODO: Does this need to be handled?
 		fn note_uncle(_author: T::AccountId, _age: T::BlockNumber) {}
 	}
+
+	impl<T: Config> pallet_session::SessionManager<T::AccountId> for Pallet<T> {
+		fn new_session(new_index: SessionIndex) -> Option<Vec<T::AccountId>> {
+			log::info!(
+				"assembling new collators for new session {} at #{:?}",
+				new_index,
+				<frame_system::Pallet<T>>::block_number(),
+			);
+
+			Self::select_top_candidates(new_index);
+
+			Some(<SelectedCandidates<T>>::get())
+		}
```

Comment:
@wischli we need to select the new collators here. Is this correct!?

Reply:
Yes, looks good to me.